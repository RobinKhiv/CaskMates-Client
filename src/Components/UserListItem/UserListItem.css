
.WhiskeyList{
  min-height: 7em;
}