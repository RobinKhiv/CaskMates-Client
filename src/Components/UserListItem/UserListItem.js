import React, { Component } from 'react';
import {Link} from 'react-router-dom';
import './UserListItem.css'
import WhiskeyApiService from '../../Services/whiskey-api-service';
import WhiskeyListContext from '../../Context/WhiskeyListContext';



export default class UserListItem extends Component {
  constructor(){
    super();
    this.state={
      showMoveMenu: false
    }
  }
  static contextType = WhiskeyListContext;

  showMoveMenu = (e) => {
    e.preventDefault();
    this.setState({showMoveMenu: true}, () => {
      document.addEventListener('mouseup', this.closeMoveMenu)
    });
  }
  
  closeMoveMenu = e => {
    e.stopPropagation();
    this.setState({showMoveMenu: false}, () => {
      document.removeEventListener('mouseup', this.closeMoveMenu)
    })
  }

  moveToList = (whiskey_Id, newlist_name, currentListID, updatedListName) => {
    WhiskeyApiService.moveToList(whiskey_Id, newlist_name, currentListID)  
    .then(() => this.context.updateWhiskeyListState(currentListID, updatedListName))
  }

  deleteFromList = (list_id) => {
    WhiskeyApiService.removeWhiskeyFromApi(list_id)
    .then(() => this.context.removeWhiskeyFromState(list_id))
  }

  render() {
    const {whiskey} = this.props;
    return (
    <div className="col-3 WhiskeyList userWhiskeys"style={{backgroundImage: `url(${whiskey.image})`}}>
      <div className='listText'>

      <Link to={`/whiskeys/${whiskey.whiskey_id}`} className='whiskeyListItem'>
      <div className="whiskeyListName">
        <h2>{whiskey.whiskeyName}</h2>
      </div>
      </Link>
      </div>
      <input type="button" name="moveWhiskey" value="Move to" onMouseDown={this.showMoveMenu}/>
      {this.state.showMoveMenu ? (
      <div className="menu" >
        <input type="button" name="FavoriteLst" onMouseUp={() => this.moveToList(whiskey.whiskey_id, 1, whiskey.id, 'Favorite List')} value="Favorite List"/>
        <input type="button" name="wishLst" onMouseUp={() => this.moveToList(whiskey.whiskey_id, 2,whiskey.id, 'Wish List')} value="Wish List"/>
        <input type="button" name="alreadyTriedLst" onMouseUp={()=> this.moveToList(whiskey.whiskey_id, 3,whiskey.id, 'Already Tried')} value="Already Tried List"/>
      </div>) : ""}
      <input type="button" name="delete" onClick={() =>this.deleteFromList(whiskey.id)}value="remove"/>
    </div>
    )
  }
}



