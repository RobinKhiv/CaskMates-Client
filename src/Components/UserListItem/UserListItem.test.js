import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import UserListItem from './UserListItem';

describe('<UserListItem />', () => {
    const whiskey = {
        whiskey_id: 1,
        image: 'https',
        whiskeyName: 'name',
    };
    it('renders without crashing', () => { 
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <UserListItem whiskey={whiskey}/>
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

  it('renders without crashing', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <UserListItem whiskey={whiskey}/>
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot(); 
  });
});