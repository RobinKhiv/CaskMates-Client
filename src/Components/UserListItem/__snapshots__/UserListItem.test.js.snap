// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserListItem /> renders without crashing 1`] = `
<div
  className="col-3 WhiskeyList userWhiskeys"
  style={
    Object {
      "backgroundImage": "url(https)",
    }
  }
>
  <div
    className="listText"
  >
    <a
      className="whiskeyListItem"
      href="/whiskeys/1"
      onClick={[Function]}
    >
      <div
        className="whiskeyListName"
      >
        <h2>
          name
        </h2>
      </div>
    </a>
  </div>
  <input
    name="moveWhiskey"
    onMouseDown={[Function]}
    type="button"
    value="Move to"
  />
  
  <input
    name="delete"
    onClick={[Function]}
    type="button"
    value="remove"
  />
</div>
`;
